### PR TITLE
feat: Add TTL recommendation for DNS zones

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

This pull request includes several changes to the `azure-resources/Network` directory, focusing on visibility settings, adding development markers, and new recommendations for DNS zones. The most important changes include making the `dnsZones` section visible, marking KQL files as under development, and adding recommendations for Time-To-Live (TTL) settings for both public and private DNS zones.

### Visibility Settings:
* [`azure-resources/Network/dnsZones/_index.md`](diffhunk://#diff-48ffb45778b76483e2d62f9c5e3bb6b8d9fdd2cd4639d2d6cb3b25ff9ac04495L4-R4): Changed `geekdocHidden` from `true` to `false` to make the `dnsZones` section visible.

### Development Markers:
* [`azure-resources/Network/dnsZones/kql/7d09523b-b3c0-403e-b104-d5d46240d683.kql`](diffhunk://#diff-aaeaad70791a60ecad40f5dd0d18c63e48790a50c9bf8d5c6b133ccd3c4f995bR1-R2): Added an under-development marker.
* [`azure-resources/Network/privateDnsZones/kql/3538aa48-c40b-455b-a93b-269fe6e65be2.kql`](diffhunk://#diff-aaeaad70791a60ecad40f5dd0d18c63e48790a50c9bf8d5c6b133ccd3c4f995bR1-R2): Added an under-development marker.

### Recommendations for DNS Zones:
* [`azure-resources/Network/dnsZones/recommendations.yaml`](diffhunk://#diff-d2979283c3a47d25fcbb6fb1b2612a12d8ebd12f24990ed78d6458ff274ad896R1-R18): Added a new recommendation to ensure TTL settings for public DNS zones meet RPO targets.
* [`azure-resources/Network/privateDnsZones/recommendations.yaml`](diffhunk://#diff-c7bb29f4b97e5cf462a5f4cdf853fc601f0922fc9176ff3fb4a64f499a915f32R57-R75): Added a new recommendation to ensure TTL settings for private DNS zones meet RPO targets.

## Related Issues/Work Items

None

### Breaking Changes

None

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Review by oZakari — DISMISSED** [review]
LGTM

File: azure-resources/Network/dnsZones/recommendations.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+- description: Ensure Time-To-Live (TTL) is set appropriately to ensure RPOs can be met
+  aprlGuid: 7d09523b-b3c0-403e-b104-d5d46240d683
+  recommendationTypeId: null
+  recommendationControl: Disaster Recovery
+  recommendationImpact: High
+  recommendationResourceType: Microsoft.Network/dnsZones
+  recommendationMetadataState: Active
+  longDescription: |
+    Azure Private DNS allows the Time-To-Live (TTL) for records in the zone to be set to a value between 1 and 2147483647 seconds. You should ensure that the TTL for the DNS records in your DNS Zones are set appropriately to meet your RPO targets to ensure that records are not cached by another system.
```
Comment (ejhenry) [inline azure-resources/Network/dnsZones/recommendations.yaml:9]:
```suggestion
    Azure DNS allows the Time-To-Live (TTL) for records in the zone to be set to a value between 1 and 2147483647 seconds. You should ensure that the TTL for the DNS records in your DNS Zones are set appropriately to meet your RPO targets to ensure that records are not cached by another system.
```

Comment (ehaslett) [reply azure-resources/Network/dnsZones/recommendations.yaml:9]: > 



> TTL for the DNS records

Our documentation talks about TTLs for record sets, so may want to re-word further:

"... TTL for DNS record sets... " or "...TTL for each record set..."

https://learn.microsoft.com/en-us/azure/dns/dns-zones-records#time-to-live

**Comment by oZakari** [comment]
@ehaslett, I've updated the guidance per your suggestion.

**Review by oZakari — DISMISSED** [review]

**Review by oZakari — DISMISSED** [review]

**Review by oZakari — APPROVED** [review]

**Review by ejhenry — APPROVED** [review]
LGTM 👍 thanks @jtracey93!

**Pull request merged**